tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,247 @@
+			embedded.WithComponentsPath("./components/consumer2"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName1, topicName1, consumerGroup1, consumerGroup2)).
+		Step("publish messages to unUsedTopic", publishMessages(metadata, sidecarName1, unUsedTopic)).
+		Step("Verify if app1 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup1)).
+		Step("Verify if app2 has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
+		Step("reset", flow.Reset(consumerGroup2)).
+
+		// Run the third application.
+		Step(app.Run(appID3, fmt.Sprintf(":%d", appPort+portOffset*2),
+			subscriberApplication(appID3, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 3.
+		Step(sidecar.Run(sidecarName3,
+			embedded.WithComponentsPath("./components/consumer3"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*2),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*2),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*2),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*2),
+			runtime.WithPubSubs(component))).
+
+		// publish message again to topic1, however this time there are more than one consumer group on same topic
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName1, topicName1, consumerGroup2)).
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName2, topicName1, consumerGroup2)).
+		Step("Verify if app1, app3 together has recevied messages published to topic1", assertMessages(10*time.Second, consumerGroup2)).
+
+		// Run the 4th application.
+		Step(app.Run(appID4, fmt.Sprintf(":%d", appPort+portOffset*3),
+			subscriberApplication(appID4, topicToBeCreated, consumerGroup4))).
+		//
+		// Run the Dapr sidecar with the component entitymanagement
+		Step(sidecar.Run(sidecarName4,
+			embedded.WithComponentsPath("./components/entitymanagementconsumer"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*3),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*3),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*3),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*3),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(metadata, sidecarName4, topicToBeCreated, consumerGroup4)).
+		Step("Verify if app5 has recevied messages published to topic", assertMessages(50*time.Second, consumerGroup4)).
+
+		// IOT Test : app 5
+		// PREREQ : Add messages to IOT endpoint via (./send-iot-device-events.sh)
```

Comment:
is it a pre req that `./send-io-device-events.sh` is run before this test ? 
Can that logic be pulled into this test and be setup in a step here?

Reply:
Is it a pre req that ./send-io-device-events.sh is run before this test ?
- Yes

Can that logic be pulled into this test and be setup in a step here?

- Yes that's one option, second option is to pull this as part of `github workflow`. Option one sounds closer to test, will give it a shot.

Reply:
to be tracked as part of : https://github.com/dapr/components-contrib/issues/1664